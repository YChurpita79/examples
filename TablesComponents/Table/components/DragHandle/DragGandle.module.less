
.HandleWrapper {
  height: 20px;
  min-width: 20px;
  vertical-align: bottom;
  display: inline-block;
  margin-right: 5px;
  svg {
    cursor: pointer;
    width: 100%;
    height: 100%;
  }
}

.selectable {
  position: absolute;
  margin-right: 0;
  left: 8px
}