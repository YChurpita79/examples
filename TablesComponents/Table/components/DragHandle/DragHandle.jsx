/* eslint-disable max-len */
import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

import styles from "./DragGandle.module.less";
import { DraggableIcon } from "icons";
const DragHandle = (props) => {
  const { isSelectable } = props;
  return (
    <div className={cn(styles.HandleWrapper, { [styles.selectable]: isSelectable })} {...props}>
      <DraggableIcon />
    </div>
  );
};

DragHandle.propTypes = {
  isSelectable: PropTypes.bool,
};
export default DragHandle;
