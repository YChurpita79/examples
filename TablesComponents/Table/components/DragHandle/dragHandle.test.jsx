import React from "react";
import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import DragHandle from "./DragHandle";

describe("DragHandle component", () => {
  it("should be rendered ", () => {
    const { container } = render(<DragHandle />);
    expect(container).toMatchSnapshot();
  });

  it("should be rendered selectable ", () => {
    const { container } = render(<DragHandle isSelectable={true} />);
    expect(container).toMatchSnapshot();
  });
});
