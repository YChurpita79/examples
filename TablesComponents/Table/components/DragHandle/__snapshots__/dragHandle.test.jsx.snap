// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DragHandle component > should be rendered  1`] = `
<div>
  <div
    class="HandleWrapper"
  >
    <svg
      fill="none"
      height="15"
      viewBox="0 0 10 15"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        id="squares"
      >
        <path
          d="M9.32324 8.86803V6.13202C9.32324 6.01108 9.2752 5.8951 9.18968 5.80959C9.10417 5.72407 8.98818 5.67603 8.86724 5.67603H6.13124C6.0103 5.67603 5.89432 5.72407 5.8088 5.80959C5.72328 5.8951 5.67524 6.01108 5.67524 6.13202V8.86803C5.67524 8.98897 5.72328 9.10494 5.8088 9.19046C5.89432 9.27598 6.0103 9.32402 6.13124 9.32402H8.86724C8.98818 9.32402 9.10417 9.27598 9.18968 9.19046C9.2752 9.10494 9.32324 8.98897 9.32324 8.86803Z"
          fill="#ADB1C9"
          id="path-1_71_"
        />
        <path
          d="M9.32324 14.544V11.808C9.32324 11.6871 9.2752 11.5711 9.18968 11.4856C9.10417 11.4001 8.98818 11.3521 8.86724 11.3521H6.13124C6.0103 11.3521 5.89432 11.4001 5.8088 11.4856C5.72328 11.5711 5.67524 11.6871 5.67524 11.808V14.544C5.67524 14.665 5.72328 14.781 5.8088 14.8665C5.89432 14.952 6.0103 15.0001 6.13124 15.0001H8.86724C8.98818 15.0001 9.10417 14.952 9.18968 14.8665C9.2752 14.781 9.32324 14.665 9.32324 14.544Z"
          fill="#ADB1C9"
          id="path-1_70_"
        />
        <path
          d="M9.32324 3.192V0.456C9.32324 0.335061 9.2752 0.219076 9.18968 0.133559C9.10417 0.0480427 8.98818 0 8.86724 0L6.13124 0C6.0103 0 5.89432 0.0480427 5.8088 0.133559C5.72328 0.219076 5.67524 0.335061 5.67524 0.456V3.192C5.67524 3.31294 5.72328 3.42892 5.8088 3.51444C5.89432 3.59996 6.0103 3.648 6.13124 3.648H8.86724C8.98818 3.648 9.10417 3.59996 9.18968 3.51444C9.2752 3.42892 9.32324 3.31294 9.32324 3.192Z"
          fill="#ADB1C9"
          id="path-1_69_"
        />
        <path
          d="M3.64746 8.86803V6.13202C3.64746 6.01108 3.59942 5.8951 3.5139 5.80959C3.42838 5.72407 3.31241 5.67603 3.19147 5.67603H0.45546C0.334521 5.67603 0.218543 5.72407 0.133026 5.80959C0.0475094 5.8951 -0.000534058 6.01108 -0.000534058 6.13202V8.86803C-0.000534058 8.98897 0.0475094 9.10494 0.133026 9.19046C0.218543 9.27598 0.334521 9.32402 0.45546 9.32402H3.19147C3.31241 9.32402 3.42838 9.27598 3.5139 9.19046C3.59942 9.10494 3.64746 8.98897 3.64746 8.86803Z"
          fill="#ADB1C9"
          id="path-1_68_"
        />
        <path
          d="M3.64746 14.544V11.808C3.64746 11.6871 3.59942 11.5711 3.5139 11.4856C3.42838 11.4001 3.31241 11.3521 3.19147 11.3521H0.45546C0.334521 11.3521 0.218543 11.4001 0.133026 11.4856C0.0475094 11.5711 -0.000534058 11.6871 -0.000534058 11.808V14.544C-0.000534058 14.665 0.0475094 14.781 0.133026 14.8665C0.218543 14.952 0.334521 15.0001 0.45546 15.0001H3.19147C3.31241 15.0001 3.42838 14.952 3.5139 14.8665C3.59942 14.781 3.64746 14.665 3.64746 14.544Z"
          fill="#ADB1C9"
          id="path-1_67_"
        />
        <path
          d="M3.64746 3.192V0.456C3.64746 0.335061 3.59942 0.219076 3.5139 0.133559C3.42838 0.0480427 3.31241 0 3.19147 0L0.45546 0C0.334521 0 0.218543 0.0480427 0.133026 0.133559C0.0475094 0.219076 -0.000534058 0.335061 -0.000534058 0.456V3.192C-0.000534058 3.31294 0.0475094 3.42892 0.133026 3.51444C0.218543 3.59996 0.334521 3.648 0.45546 3.648H3.19147C3.31241 3.648 3.42838 3.59996 3.5139 3.51444C3.59942 3.42892 3.64746 3.31294 3.64746 3.192Z"
          fill="#ADB1C9"
          id="path-1_66_"
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`DragHandle component > should be rendered selectable  1`] = `
<div>
  <div
    class="HandleWrapper selectable"
  >
    <svg
      fill="none"
      height="15"
      viewBox="0 0 10 15"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        id="squares"
      >
        <path
          d="M9.32324 8.86803V6.13202C9.32324 6.01108 9.2752 5.8951 9.18968 5.80959C9.10417 5.72407 8.98818 5.67603 8.86724 5.67603H6.13124C6.0103 5.67603 5.89432 5.72407 5.8088 5.80959C5.72328 5.8951 5.67524 6.01108 5.67524 6.13202V8.86803C5.67524 8.98897 5.72328 9.10494 5.8088 9.19046C5.89432 9.27598 6.0103 9.32402 6.13124 9.32402H8.86724C8.98818 9.32402 9.10417 9.27598 9.18968 9.19046C9.2752 9.10494 9.32324 8.98897 9.32324 8.86803Z"
          fill="#ADB1C9"
          id="path-1_71_"
        />
        <path
          d="M9.32324 14.544V11.808C9.32324 11.6871 9.2752 11.5711 9.18968 11.4856C9.10417 11.4001 8.98818 11.3521 8.86724 11.3521H6.13124C6.0103 11.3521 5.89432 11.4001 5.8088 11.4856C5.72328 11.5711 5.67524 11.6871 5.67524 11.808V14.544C5.67524 14.665 5.72328 14.781 5.8088 14.8665C5.89432 14.952 6.0103 15.0001 6.13124 15.0001H8.86724C8.98818 15.0001 9.10417 14.952 9.18968 14.8665C9.2752 14.781 9.32324 14.665 9.32324 14.544Z"
          fill="#ADB1C9"
          id="path-1_70_"
        />
        <path
          d="M9.32324 3.192V0.456C9.32324 0.335061 9.2752 0.219076 9.18968 0.133559C9.10417 0.0480427 8.98818 0 8.86724 0L6.13124 0C6.0103 0 5.89432 0.0480427 5.8088 0.133559C5.72328 0.219076 5.67524 0.335061 5.67524 0.456V3.192C5.67524 3.31294 5.72328 3.42892 5.8088 3.51444C5.89432 3.59996 6.0103 3.648 6.13124 3.648H8.86724C8.98818 3.648 9.10417 3.59996 9.18968 3.51444C9.2752 3.42892 9.32324 3.31294 9.32324 3.192Z"
          fill="#ADB1C9"
          id="path-1_69_"
        />
        <path
          d="M3.64746 8.86803V6.13202C3.64746 6.01108 3.59942 5.8951 3.5139 5.80959C3.42838 5.72407 3.31241 5.67603 3.19147 5.67603H0.45546C0.334521 5.67603 0.218543 5.72407 0.133026 5.80959C0.0475094 5.8951 -0.000534058 6.01108 -0.000534058 6.13202V8.86803C-0.000534058 8.98897 0.0475094 9.10494 0.133026 9.19046C0.218543 9.27598 0.334521 9.32402 0.45546 9.32402H3.19147C3.31241 9.32402 3.42838 9.27598 3.5139 9.19046C3.59942 9.10494 3.64746 8.98897 3.64746 8.86803Z"
          fill="#ADB1C9"
          id="path-1_68_"
        />
        <path
          d="M3.64746 14.544V11.808C3.64746 11.6871 3.59942 11.5711 3.5139 11.4856C3.42838 11.4001 3.31241 11.3521 3.19147 11.3521H0.45546C0.334521 11.3521 0.218543 11.4001 0.133026 11.4856C0.0475094 11.5711 -0.000534058 11.6871 -0.000534058 11.808V14.544C-0.000534058 14.665 0.0475094 14.781 0.133026 14.8665C0.218543 14.952 0.334521 15.0001 0.45546 15.0001H3.19147C3.31241 15.0001 3.42838 14.952 3.5139 14.8665C3.59942 14.781 3.64746 14.665 3.64746 14.544Z"
          fill="#ADB1C9"
          id="path-1_67_"
        />
        <path
          d="M3.64746 3.192V0.456C3.64746 0.335061 3.59942 0.219076 3.5139 0.133559C3.42838 0.0480427 3.31241 0 3.19147 0L0.45546 0C0.334521 0 0.218543 0.0480427 0.133026 0.133559C0.0475094 0.219076 -0.000534058 0.335061 -0.000534058 0.456V3.192C-0.000534058 3.31294 0.0475094 3.42892 0.133026 3.51444C0.218543 3.59996 0.334521 3.648 0.45546 3.648H3.19147C3.31241 3.648 3.42838 3.59996 3.5139 3.51444C3.59942 3.42892 3.64746 3.31294 3.64746 3.192Z"
          fill="#ADB1C9"
          id="path-1_66_"
        />
      </g>
    </svg>
  </div>
</div>
`;
